pkg/infrastructure/clusterapi/clusterapi.go

```diff
@@ -289,38 +285,34 @@ func (i *InfraProvider) Provision(ctx context.Context, dir string, parents asset
 		timezone, _ := untilTime.Zone()
 		reqBootstrapPubIP := installConfig.Config.Publish == types.ExternalPublishingStrategy && i.impl.BootstrapHasPublicIP()
 		logrus.Infof("Waiting up to %v (until %v %s) for machines %v to provision...", timeout, untilTime.Format(time.Kitchen), timezone, machineNames)
-		if err := wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
-			Duration: time.Second * 10,
-			Factor:   float64(1.5),
-			Steps:    32,
-			Cap:      timeout,
-		}, func(ctx context.Context) (bool, error) {
-			allReady := true
-			for _, machineName := range machineNames {
-				machine := &clusterv1.Machine{}
-				if err := cl.Get(ctx, client.ObjectKey{
-					Name:      machineName,
-					Namespace: capiutils.Namespace,
-				}, machine); err != nil {
-					if apierrors.IsNotFound(err) {
-						logrus.Debugf("Not found")
-						return false, nil
+		if err := wait.PollUntilContextTimeout(ctx, 15*time.Second, timeout, true,
+			func(ctx context.Context) (bool, error) {
+				allReady := true
+				for _, machineName := range machineNames {
+					machine := &clusterv1.Machine{}
+					if err := cl.Get(ctx, client.ObjectKey{
+						Name:      machineName,
+						Namespace: capiutils.Namespace,
+					}, machine); err != nil {
+						if apierrors.IsNotFound(err) {
+							logrus.Debugf("Not found")
```

Comment:
nit: I know this is just an indent change, but while we are here can we drop this or add a bit more info to what wasn't found?

Reply:
We shouldn't drop it,. because this protects from failing the install if we check for the machine too quickly. We could add the name of the machine that isn't found, but I would rather not push a new commit and rerun all the tests for this change.